db/cql_type_parser.cc

```diff
@@ -37,11 +37,27 @@ data_type db::cql_type_parser::parse(const sstring& keyspace, const sstring& str
     if (i != native_types.end()) {
         return i->second.get_type();
     }
+    return std::nullopt;
+}
 
+data_type db::cql_type_parser::parse(const sstring& keyspace, const sstring& str, const data_dictionary::user_types_storage& uts) {
+    auto native = parse_native_type(str);
+    if (native) {
+        return *native;
+    }
     auto raw = parse_raw(str);
     return raw->prepare_internal(keyspace, uts.get(keyspace)).get_type();
 }
 
+data_type db::cql_type_parser::parse(const sstring& keyspace, const sstring& str, const data_dictionary::user_types_metadata& utm) {
+    auto native = parse_native_type(str);
+    if (native) {
+        return *native;
+    }
+    auto raw = parse_raw(str);
+    return raw->prepare_internal(keyspace, utm).get_type();
```

Comment:
This was my initial approach, but because of this, I was getting an error in some seemingly unrelated test, because it used a non-existent keyspace for the `parse` call - in the current code, it short-circuited on the native types tests, but using the one-liner we got an error on `uts.get(keyspace)`.
But now that I look at it, it's probably the test that should have been fixed, and not the change - I used this approach in the rebase